src/partition/ReceivedPartition.cpp

```diff
@@ -262,6 +263,13 @@ void ReceivedPartition::compute()
     PRECICE_ASSERT(_mesh->getVertexOffsets().empty());
     _mesh->setVertexOffsets(std::move(vertexOffsets));
   }
+
+  PRECICE_ASSERT(!_mesh->getVertexOffsets().empty());
+  /* TODO this triggers even though it shouldn't. Asserting in the branch above after setting it does not trigger
+  if (m2n().usesTwoLevelInitialization() && utils::IntraComm::isPrimary()) {
+    PRECICE_ASSERT(!_mesh->getVertexDistribution().empty());
+  }
```

Comment:
This is concerning.

This assertion works fine after l227. So either there is some UB leading so nonsense or my condition evaluates to something else.

Reply:
Wrong condition. This only happens without 2L1.